bookshelf/src/controllers.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Samples\Bookshelf;
+
+/*
+ * Adds all the controllers to $app.  Follows Silex Skeleton pattern.
+ */
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+$listBooks = function (Request $request) use ($app) {
+    /** @var DataModel $model */
+    $model = $app['bookshelf.model'];
+    /** @var Twig_Environment $twig */
+    $twig = $app['twig'];
+    $token = $request->query->get('page_token');
+    $bookList = $model->listBooks($app['bookshelf.page_size'], $token);
+
+    return $twig->render('list.html', array(
+        'books' => $bookList['books'],
+        'next_page_token' => $bookList['cursor'],
+    ));
+};
+$app->get('/', $listBooks);
+$app->get('/books/', $listBooks);
+
+$app->get('/books/add', function () use ($app) {
+    /** @var Twig_Environment $twig */
+    $twig = $app['twig'];
+
+    return $twig->render('form.html', array(
+        'action' => 'Add',
+        'book' => array(),
+    ));
+});
+
+$app->post('/books/add', function (Request $request) use ($app) {
+    /** @var DataModel $model */
+    $model = $app['bookshelf.model'];
+    /** @var ImageStorage $storage */
```

Comment:
Do these typehints automatically pick up the parent namespace? If so, I need to fix all my samples :dizzy_face: